components/tidb_query_executors/src/runner.rs

```diff
@@ -510,6 +508,16 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
         self.out_most_executor.collect_storage_stats(dest);
     }
 
+    pub fn collect_scan_summary(&mut self, dest: &mut ExecSummary) {
+        if !self.collect_exec_summary {
+            return;
+        }
+        let result = &self.exec_stats.summary_per_executor;
+        if let Some(exec_stat) = result.first() {
+            dest.clone_from(exec_stat);
+        }
```

Comment:
I think can't work here. Have you tested it?

Reply:
I have tested it just in TiDB. (Print the log about the KVReadWallTimeMS in copr(select_result.go) logic, and the result show me that it's not the zero🙈)